src/picongpu/include/fields/background/templates/TWTS/RotateField.tpp

```diff
@@ -0,0 +1,127 @@
+    
+    template <typename T_Type, typename T_AngleType>
+    struct RotateField;
+    
+    template <typename T_Type, typename T_AngleType>
+    struct RotateField<PMacc::math::Vector<T_Type,3>, T_AngleType >
+    {
+        typedef PMacc::math::Vector<T_Type,3> result;
+        typedef T_AngleType AngleType;
+        HDINLINE result
+        operator()( const result& fieldPosVector,
+                    const AngleType phi ) const
+        {
+            /*  Since, the laser propagation direction encloses an angle of phi with the
+             *  simulation y-axis (i.e. direction of sliding window), the positions vectors are
+             *  rotated around the simulation x-axis before calling the TWTS field functions.
+             *  Note: The TWTS field functions are in non-rotated frame and only use the angle
+             *  phi to determine the required amount of pulse front tilt.
+             *  RotationMatrix[PI/2+phi].(y,z) (180Deg-flip at phi=90Deg since coordinate
+             *  system in paper is oriented the other way round.) */
+            return result(
+                fieldPosVector.x(),
+               -pmMath::sin(AngleType(phi))*fieldPosVector.y()
+                    -pmMath::cos(AngleType(phi))*fieldPosVector.z() ,
+               +pmMath::cos(AngleType(phi))*fieldPosVector.y()
+                    -pmMath::sin(AngleType(phi))*fieldPosVector.z() );
+        }
+                        
+    };
+    
+    template <typename T_Type, typename T_AngleType>
+    struct RotateField<PMacc::math::Vector<T_Type,2>, T_AngleType >
+    {
+        typedef PMacc::math::Vector<T_Type,2> result;
+        typedef T_AngleType AngleType;
+        HDINLINE result
+        operator()( const result& fieldPosVector,
+                    const AngleType phi ) const
+        {
+            /*  Since, the laser propagation direction encloses an angle of phi with the
+             *  simulation y-axis (i.e. direction of sliding window), the positions vectors are
+             *  rotated around the simulation x-axis before calling the TWTS field functions.
+             *  Note: The TWTS field functions are in non-rotated frame and only use the angle
+             *  phi to determine the required amount of pulse front tilt.
+             *  RotationMatrix[PI/2+phi].(y,z) (180Deg-flip at phi=90Deg since coordinate
+             *  system in paper is oriented the other way round.) */
+            
+            /*  Rotate 90 degree around y-axis, so that TWTS laser propagates within
+             *  the 2D (x,y)-plane. Corresponding position vector for the Ez-components
+             *  in 2D simulations.
+             *  3D     3D vectors in 2D space (x,y)
+             *  x -->  z
+             *  y -->  y
+             *  z --> -x (Since z=0 for 2D, we use the existing
+             *            TWTS-field-function and set -x=0)
+             *  Ex --> Ez (--> Same function values can be used in 2D,
```

Comment:
I do not understand what `Ex --> Ez (--> Same function values can be used in 2D but with Yee-Cell-Positions for Ez.)` shall tell me and why it is specific to `Yee` since PIConGPU supports arbitrary positions for fields in cells, not only Yee staggering.